### ISSUE TITLE
Nixus stores meaningful logs in the remotes

### ISSUE THREAD

**Issue by lovesegfault** [Infinisil/nixus#9]
When a nixus deployment fails inspecting the local log in `/var/lib/system-switcher` is almost useless as all of the meaningful, informative, end up in the _remote's_ `/var/lib/system-switcher`. 

It'd be nice if `nixus` copied over the failure logs.

**Comment by Infinisil** [comment]
Just pushed https://github.com/Infinisil/nixus/commit/c62336b9b0a8eed0ab8503cbfd2869cfc243389d for a temporary fix for less confusion

**Comment by bqv** [comment]
Implemented this while doing my execline fiddling. Works as follows:

[Create a fifo as current/fifo to act as the pipe](https://github.com/bqv/nixus/blob/46de0b0202654713cd5fdef4b9be2bfdeb856e9f/scripts/switch#L129)

[Open FD3 to write to the fifo](https://github.com/bqv/nixus/blob/46de0b0202654713cd5fdef4b9be2bfdeb856e9f/scripts/switch#L130)

[Dup FD1 (local logs) to FD3 so activation output is copied to the fifo](https://github.com/bqv/nixus/blob/46de0b0202654713cd5fdef4b9be2bfdeb856e9f/scripts/switch#L131)

[Set a background process to cat the fifo over ssh into STDOUT inherited from the main progress (ended by EOF from the remote)](https://github.com/bqv/nixus/blob/46de0b0202654713cd5fdef4b9be2bfdeb856e9f/modules/deploy.nix#L138-L149)

[To save littering and help ensure EOF is sent when the process ends (no other open writers), delete the fifo when done](https://github.com/bqv/nixus/blob/46de0b0202654713cd5fdef4b9be2bfdeb856e9f/scripts/switch#L154)

**Comment by lheckemann** [comment]
> Dup FD1 (local logs) to FD3 so activation output is copied to the fifo

AFAIU all the output will be sent to the FIFO only with this approach (you're duplicating FD3 to FD1, not duplicating the output to both FDs). I think `tee` should be used instead, in bash I'd do this using `exec > >(tee /path/to/fifo)`.

**Comment by bqv** [comment]
@lheckemann not all the echoes in the switch script go to FD1, some of them are echoed to FD2. Also i am duplicating FD3 to FD1, that's the behaviour of n>&m as I understand it (and experimentally verified) but it is bash, so I wouldn't be surprised if this doesn't work on full moons.

I very very intentionally avoided using tee, because this really doesn't need yet another random process hanging around with the potential to become an orphan, just to open a named pipe

**Comment by lheckemann** [comment]
Right, I think I may have misunderstood "Dup FD1 (local logs) to FD3 so activation output is copied to the fifo" then — I thought you wanted all output to be sent both to the original stdout _and_ to the FIFO. But if you only want it in the FIFO all is well :)

**Comment by bqv** [comment]
Ah, no, that's not really necessary. There's already the initial tee copying the output to the log, and the original stdout should be a nohup'd black hole at this point anyway. I'll probably make that more obvious when I turn the bash in my version's switch script into execline script too
